src/Parser.php

```diff
@@ -362,7 +365,7 @@ private function filterBodyNodes(Node $node, bool $nested = false): ?Node
         // we need to discard the wrapping "Node" for the "body" node
         $nested = $nested || !$node instanceof Nodes;
         foreach ($node as $k => $n) {
-            if (null !== $n && null === $this->filterBodyNodes($n, $nested)) {
```

Comment:
node is `@implements \IteratorAggregate<int|string, Node>` so `$n` is never null.